### PR TITLE
[BIO] 21-4192 PR3 PDF generation

### PR DESCRIPTION
**Pull request by voidspooks**
## Summary

- **This work is behind a feature toggle (flipper): NO** (feature flag will be added in PR4)
- This PR implements the PDF generation layer for VA Form 21-4192 (Request for Employment Information in Connection with Claim for Disability Benefits):
  - **VA214192 PDF Form Mapping Class**: 480 lines of field mappings that translate form data to the official VA PDF template
  - **21-4192.pdf Template**: Official VA form template (binary file)
  - **PdfFill::Filler Registration**: Registers the form in the PDF generation system
  - **Comprehensive PDF Specs**: 147 lines of tests covering all form sections and field mappings
- **What does this do?**: Takes the structured form data from the model (PR2) and generates a filled PDF that matches the official VA Form 21-4192 template, ready for submission to VA systems.
- **Team**: Benefits Intake Optimization team owns the maintenance of this component.
- **Why this approach?**: Follows the established `PdfFill::Forms` pattern used throughout vets-api for PDF generation, ensuring consistency with other VA forms (21-526EZ, 21-0966, etc.).

## Related issue(s)

- Part of Form 21-4192 MVP implementation
- Epic: VA Form 21-4192 Implementation
- **Depends on**: PR1 and PR2 must be merged first

## Testing done

- [x] **New code is covered by unit tests** - 147 lines of comprehensive PDF generation specs
- **Old behavior**: Form 21-4192 PDF generation did not exist in the system
- **New behavior**: 
  - System can generate filled PDFs from form data
  - All form sections are properly mapped to PDF fields:
    - Veteran Information (name, DOB, SSN, address, contact)
    - Employment Information (employer details, dates, duties, hours worked)
    - Military Duty Status
    - Benefit Entitlement Payments
  - Date formatting, address formatting, and field truncation handled correctly
  - Checkbox and text field mappings work as expected
- **Verification steps**:
  1. Run PDF specs: `bundle exec rspec spec/lib/pdf_fill/forms/va214192_spec.rb`
  2. Verify all field mappings:
     - Veteran name fields map correctly (first, middle, last)
     - Address fields handle multi-line addresses
     - Date fields format correctly (MM/DD/YYYY)
     - Employment information maps to correct PDF positions
     - Checkboxes render properly for yes/no fields
  3. Verify PDF generation:
     - Valid form data produces a filled PDF
     - Missing optional fields don't break generation
     - Field truncation prevents overflow
  4. All linting and CI checks pass
- **No flipper in this PR** - the feature flag will be added in PR4 with the controller

## Screenshots
Screenshot 2025-10-22 at 11 51 15 AM

## What areas of the site does it impact?

- **New functionality**: Adds Form 21-4192 PDF generation (no existing functionality modified)
- **PDF System**: Registers new form in `PdfFill::Filler` alongside existing forms
- **Dependencies**: 
  - Requires PR1 bug fixes for proper error handling
  - Requires PR2 model for form data structure
- **Future impact**: This PDF generator will be used by the controller (PR4) to create downloadable PDFs

## Acceptance criteria

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No error nor warning in the console.
- [x] Events are being sent to the appropriate logging solution
- [x] Documentation has been updated (link to documentation) - _Class includes inline documentation for field mappings_
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Feature/bug has a monitor built into Datadog (if applicable) - _Monitoring will be added with controller in PR4_
- [x] If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected - _No routes in this PR_
- [x] I added a screenshot of the developed feature - _N/A for PDF generation logic_

## Requested Feedback

**Key Points for Review:**

1. **Large File Size**: This PR is ~629 lines, slightly over the 500-line guideline. However, this cannot be split further because:
   - The PDF form mapping class (`va214192.rb` - 480 lines) is a single cohesive unit that maps all form fields
   - Splitting the mapping from its tests would violate the principle of keeping code and tests together
   - The mapping follows the same pattern as other large form mappings in the codebase (e.g., 21-526EZ)

2. **PDF Field Mapping Strategy**: 
   - Each form section (veteran info, employment info, etc.) has dedicated methods
   - Field names match the official VA PDF form field identifiers
   - Handles edge cases: date formatting, address overflow, field truncation
   - Uses `split_checkbox` helper for yes/no fields (consistent with other forms)

3. **Test Coverage**: 147 lines of specs cover:
   - All major form sections and field mappings
   - Date formatting and address handling
   - Edge cases and missing data scenarios
   - PDF generation success/failure paths

### CODE REVIEW

File: lib/pdf_fill/forms/va214192.rb [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        return unless benefits['grossMonthlyAmountOfBenefit']
+
+        amount = benefits['grossMonthlyAmountOfBenefit'].to_f
+        dollars = amount.floor
+        cents = ((amount - dollars) * 100).round
+
+        thousands = (dollars / 1000).floor
+        hundreds = dollars % 1000
+
+        @form_data['benefitEntitlementPayments']['grossMonthlyAmountOfBenefit'] = {
```
[the remark this answers is not in the document]
Comment (voidspooks) [reply lib/pdf_fill/forms/va214192.rb]: If it was repeated more than twice, this would have more merit. Not a big deal.

File: lib/pdf_fill/forms/va214192.rb [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+
+        date = Date.parse(date_string.to_s)
+        {
+          month: date.month.to_s.rjust(2, '0'),
+          day: date.day.to_s.rjust(2, '0'),
+          year: date.year.to_s
+        }
+      rescue ArgumentError
+        nil
+      end
```
[the remark this answers is not in the document]
Comment (voidspooks) [reply lib/pdf_fill/forms/va214192.rb]: Look at my response to SSN question.

**Comment by rmtolmach** [comment]
>  This PR is ~629 lines, slightly over the 500-line guideline.

FYI: Specs get excluded from the count, along with [some other files](https://github.com/department-of-veterans-affairs/vets-api/blob/master/Dangerfile#L54-L58).

**Comment by rmtolmach** [comment]
oops–I approved without your teammate re-reviewing after the changes. Run the new changes by @LinnJS when you get a chance.

**Review by LinnJS — APPROVED** [review]
